apps/miniapp/domains/common/utils/oidcAuth.tsx

```diff
@@ -40,7 +51,54 @@ const OidcAuthProvider = ({ children }) => {
             }
         }
 
-    }, [isAllLoaded, user, isAuthenticated, condoUser, condoOrganization])
+    }, [isAllLoaded, user, isAuthenticated, condoUser, condoOrganization, hasStorageAccess])
+
+    useEffect(() => {
+        if (!hasStorageAccess && document && document.hasStorageAccess) {
```

Comment:
looks like you can reuse `requestAccessToStorage` here and skip code duplication

Reply:
There are different functions to check: `requestStorageAccess` and `hasStorageAccess`